consensus/src/consensus/inner/commit.rs

```diff
@@ -105,11 +105,18 @@ impl ConsensusInner {
                                 metrics::gauge!(metrics::blocks::HEIGHT, canon.block_height as f64);
                             }
 
-                            for block_hash in fork_path.path {
-                                if &block_hash == hash {
-                                    self.verify_and_commit_block(hash, block).await?
+                            self.storage.recommit_blockchain(&fork_path.path[0]).await?;
+                            let committed_blocks =
+                                self.storage.canon().await?.block_height as u32 - canon_branch_number;
+                            if committed_blocks > 0 && self.recommit_taint.is_none() {
+                                self.recommit_taint = Some(canon_branch_number);
+                            }
+
+                            for block_hash in &fork_path.path[committed_blocks as usize..] {
```

Comment:
are we certain `committed_blocks` is always within bounds? otherwise, `.iter().skipcommitted_blocks as usize)` could be used

Reply:
added a check, but yes.